File: src/org/labrad/qubits/config/SetupPacket.java
package org.labrad.qubits.config;

import org.labrad.data.Data;

public class SetupPacket {
  private String state;
  private Data records;

  public SetupPacket(String state, Data records) {
    this.state = state;
    this.records = records;
  }

  public String getState() { return state; }
  public Data getRecords() { return records; }
}

File: QubitServer/src/main/scala/org/labrad/qubits/config/SetupPacket.scala
package org.labrad.qubits.config

import org.labrad.data._

case class SetupPacket(state: String, records: Seq[(String, Data)]) {
  def recordData: Data = {
    val recordClusters = records.map {
      case (setting, Data.NONE) => Cluster(Str(setting)) // Can't include NONEs inside larger structures. TODO: fix this
      case (setting, data) => Cluster(Str(setting), data)
    }
    Cluster(recordClusters: _*)
  }
}
